src/strategies/hopr-uni-lp-farm/index.ts

```diff
@@ -0,0 +1,178 @@
+          internalType:"address",
+          name:"",
+          type:"address"
+       }
+    ],
+    name:"liquidityProviders",
+    outputs:[
+       {
+          internalType:"uint256",
+          name:"claimedUntil",
+          type:"uint256"
+       },
+       {
+          internalType:"uint256",
+          name:"currentBalance",
+          type:"uint256"
+       }
+    ],
+    stateMutability:"view",
+    type:"function"
+ }
+];
+
+const XDAI_BLOCK_SUBGRAPH_URL = 'https://api.thegraph.com/subgraphs/name/1hive/xdai-blocks';
+const HOPR_XDAI_SUBGRAPH_URL = 'https://api.thegraph.com/subgraphs/name/hoprnet/hopr-on-xdai';
+
+async function getXdaiBlockNumber (timestamp: number): Promise<number> {
+  const query = {
+    blocks: {
+      __args: {
+        first: 1,
+        orderBy: 'number',
+        orderDirection: 'desc',
+        where: {
+          timestamp_lte: timestamp
+        }
+      },
+      number: true,
+      timestamp: true
+    }
+  };
+  const data = await subgraphRequest(XDAI_BLOCK_SUBGRAPH_URL, query);
+  return Number(data.blocks[0].number);
+}
+
+async function xHoprSubgraphQuery (addresses: string[], blockNumber: number): Promise<{[propName: string]:number}> {
+  const query = {
+    accounts: {
+      __args: {
+        block: {
+          number: blockNumber
+        },
+        where: {
+          id_in: addresses.map(adr => adr.toLowerCase())
+        }
+      },
+      id: true,
+      totalBalance: true
+    }
+  }
+  const data = await subgraphRequest(HOPR_XDAI_SUBGRAPH_URL, query);
+  // map result (data.accounts) to addresses
+  const entries = data.accounts.map(d => [d.id, Number(d.totalBalance)])
+  return Object.fromEntries(entries);
+}
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const res = await multicall(
+    network,
+    provider,
+    tokenAndPoolAbi,
+    [
+      [options.hoprAddress, 'balanceOf', [options.uniPoolAddress]],
+      [options.uniPoolAddress, 'totalSupply', []],
+    ].concat(
+      addresses.map(
+        (address: any) => [options.uniPoolAddress, 'balanceOf', [address]]
+      )
+    ).concat(
+      blockTag >= options.farmDeployBlock || blockTag === 'latest' ? addresses.map(
+        (address: any) => [options.farmAddress, 'liquidityProviders', [address]]
+      ) : []
+    ),
+    { blockTag }
+  );
+
+  const hoprBalanceOfPool = res[0];
+  const poolTotalSupply = res[1];
+  const response: BigNumber[] = blockTag >= options.farmDeployBlock || blockTag === 'latest'
+    ? res.slice(2, 2 + addresses.length).map((r, i) => (r[0] as BigNumber).add(res[2 + i + addresses.length][1] as BigNumber))
+    : res.slice(2).map(r => r[0] as BigNumber);
+
+  // get block timestamp to search on xDai subgraph
+  const snapshotTimestamp = (await provider.getBlock(blockTag)).timestamp;
+  const snapshotXdaiBlock = await getXdaiBlockNumber(snapshotTimestamp);
```

Comment:
You could use Promise.all here to send request in same time and make strategy a bit faster to resolve, like this:
```js
const { snapshotTimestamp, snapshotXdaiBlock } = await Promise.all([
  provider.getBlock(blockTag),
  getXdaiBlockNumber(snapshotTimestamp)
]);
```

Reply:
Actually L.152, L.153 and L.155 cannot be merged within a `Promise.all`, because each async call needs value returned from precedent async call.